Pages/EditCharityEvent.razor

```diff
@@ -126,34 +108,74 @@
     KarmaContext KarmaContext = new KarmaContext();
     public CharityEvent CharityEvent;
     private string FilterValue { get; set; } = "";
-    private String nameSurname { get; set; } = "";
-    private String surnameName { get; set; } = "";
+    private string nameSurname { get; set; } = "";
+    private string surnameName { get; set; } = "";
+    private string ErrorMessage { get; set; } = "";
+    private int VolunteerCount { get; set; }
     bool panelOpenState1 = true;
     bool panelOpenState2 = true;
 
     string CurrentUserId { get; set; }
 
+    private void OnFocusHandler(FocusEventArgs e)
+    {
+        ErrorMessage = "";
+    }
+
+    private void OnFocusOutHandler(FocusEventArgs e)
+    {
+        var value = VolunteerCount;
+        if (value > 0 && value >= CharityEvent.Volunteers.Count)
+        {
+            CharityEvent.MaxVolunteers = value;
+        }
+        else if (value < 0)
+        {
+            ErrorMessage = "Illegal volunteer count, please choose a positive number.";
+        }
+        else
+        {
+            ErrorMessage = "Illegal volunteer count, please choose a value grater than currently assigned volunteeer count or reduce the number of volunteers in this event.";
+        }
+    }
+
     private void UpdateEvent()
     {
-        KarmaContext.Events.Update(CharityEvent);
-        KarmaContext.SaveChanges();
-        m_uriHelper.NavigateTo("");
-        m_notifactionTransmitter.ShowMessage("The event has been updated");
+        if (ObjectChecker.IsAnyNullOrEmpty(CharityEvent))
+        {
+            m_notifactionTransmitter.ShowMessage("There are some empty fields", MatToastType.Danger);
+        }
+        else
+        {
+            Thread thread1 = new Thread(() =>
+            {
+                KarmaContext.Events.Update(CharityEvent);
+                KarmaContext.SaveChanges();
+            });
+            Thread thread2 = new Thread(() =>
+            {
+                m_uriHelper.NavigateTo("");
+                m_notifactionTransmitter.ShowMessage("The event has been updated", MatToastType.Success);
+            });
+            thread1.Start();
+            thread2.Start();
```

Comment:
wellll since we have the requirements its fine, but you cannot say "The event has been updated" if the update is happening in the other thread and you dont know what is happening there.

Reply:
resource sharing(updateSuccess). One thread tries to update database, other waits for the first thread to complete (thread1.Join())